tools/import-wraps.py

```diff
@@ -30,19 +31,19 @@ def get_wrap_list():
     stdout = subprocess.check_output(['meson', 'wrap', 'list'])
     return stdout.decode().splitlines()
 
-def get_wrap_info(wrap):
-    versions = []
+def get_wrap_info(wrap: str):
+    versions: T.List[T.Tuple[str, int]] = []
     try:
         stdout = subprocess.check_output(['meson', 'wrap', 'info', wrap])
         for line in stdout.decode().splitlines()[1:]:
             line = line.strip()
             version, revision = line.split()
-            versions.append((version, revision))
+            versions.append((version, int(revision)))
```

Comment:
This was made because Meson itself uses revisions as ints, so it made sense to be consistent.

Reply:
Meson doesn't. When it parsed the v1 API, it did, but this was in order to compare it with the wrapdb API itself, which returned revision in a separate field as a standalone int.

In the v2 API porting work, there was a bug that it continued to cast the local version's revision to an int, in order to compare it to the remote version which was no longer automatically an int due to a side effect of the wrapdb field layout.

That got fixed in https://github.com/mesonbuild/meson/commit/38db4602d8f80c3d2aec80ed527e3eed3c41b08c by only using revisions as strings everywhere.

In this script, the revision is only ever used for building f-strings, and making it an int feels a bit inconsistent. both with its intended use and with current git versions of meson.

Reply:
Time to go change my Meson PR....